src/Bridge/Doctrine/Persister/ObjectManagerPersister.php

```diff
@@ -27,35 +27,45 @@ class ObjectManagerPersister implements PersisterInterface
 {
     use IsAServiceTrait;
 
-    private $objectManager;
+    /** @var ObjectManager[]  */
+    private $managers;
 
     /**
-     * @var array|null Values are FQCN of persistable objects
+     * @var ClassMetadata[] Entity metadata, FQCN being the key
      */
-    private $persistableClasses;
+    private $metadata = [];
 
     /**
-     * @var ClassMetadata[] Entity metadata, FQCN being the key
+     * @var string[ObjectManager] Keys are FQCN of persistable objects, Values are managers
      */
-    private $metadata = [];
+    private $classManagerMap;
 
-    public function __construct(ObjectManager $manager)
+    public function __construct(ObjectManager ...$managers)
     {
-        $this->objectManager = $manager;
+        $this->managers = $managers;
+
+        foreach ($managers as $manager) {
+            $allMetadata = $manager->getMetadataFactory()->getAllMetadata();
```

Comment:
loading all metadata of a manager is bad for performance, as this is not an optimized API. The optimized API is the one getting metadata for a given class (as the class name is the cache key).

It would be better to build the map on demand, when being asked for a specific class, rather than trying to build it upfront